src/main/java/com/meta/cp4m/configuration/RootConfiguration.java

```diff
@@ -137,7 +144,21 @@ private <T extends Message> Service<T> createService(
     } else {
       store = new NullStore<T>();
     }
-    return new Service<>(store, handler, plugin, serviceConfig.webhookPath());
+
+    PreProcessor<T> preProcessor;
+    List<PreProcessor<T>> preProcessorsList = new ArrayList<>();
+
+    if(!serviceConfig.preProcessors().isEmpty()){
+      List<String> preProcessorNames = serviceConfig.preProcessors();
+      for (String i : preProcessorNames) {
+        preProcessor = preProcessors.get(i).toPreProcessor();
+        preProcessorsList.add(preProcessor);
+      }
+
+      return new Service<>(store, handler, plugin, preProcessorsList, serviceConfig.webhookPath());
+    } else {
+      return new Service<>(store, handler, plugin, serviceConfig.webhookPath());
+    }
```

Comment:
1. we don't need the if/else here, the `Service` constructor should be completely ok with an empty list or preprocessors 
2. what happens if the preprocessor name (`i`) is not in the preprocessors? It will return null, we should throw a sane error